lib/galaxy/jobs/command_factory.py

```diff
@@ -290,10 +289,12 @@ def capture_stdout_stderr(self, stdout_file, stderr_file):
         self.append_command(f"> '{stdout_file}' 2> '{stderr_file}'",
                             sep="")
 
-    def capture_return_code(self):
+    def capture_return_code(self, exit_code_path=None):
         if not self.return_code_captured:
             self.return_code_captured = True
             self.append_command(CAPTURE_RETURN_CODE)
+            if exit_code_path:
+                self.append_command(f"echo $return_code > {exit_code_path}")
```

Comment:
Now that this is called only from one place, we can simplify this?

```suggestion
    def capture_return_code(self, exit_code_path):
        self.append_command(CAPTURE_RETURN_CODE)
        self.append_command(f"echo $return_code > {exit_code_path}")
```